language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/dependencygraph/DependencyGraph.scala

```diff
@@ -45,4 +46,24 @@ private[codegen] object DependencyGraph {
     Graph.cyclicDependencies(internalNodes = typeDeclNodes, roots = templateNodes)
   }
 
+  /** Computes the collection of templates in the `library` and
+    * all the type declarations for which code must be generated
+    * so that the output of the codegen compiles while only
+    * targeting template definitions that can be observed through
+    * the Ledger API.
+    */
+  def transitiveClosure(decls: Map[Identifier, InterfaceType]): TransitiveClosure = {
+    val dependencies = orderedDependencies(decls)
+    val (templateIds, typeDeclarations) =
+      dependencies.deps
+        .partitionMap { case (templateId, Node(content, _, _)) =>
+          content.bimap((templateId, _), (templateId, _))
+        }
+    TransitiveClosure(
+      templateIds = templateIds,
+      typeDeclarations = typeDeclarations,
+      errors = dependencies.errors,
+    )
+  }
```

Comment:
Re: https://github.com/digital-asset/daml/issues/13764

This code has been moved here and changed slightly. It seems innocuous as well but another change worth looking into, I believe.